Magento2/Sniffs/Templates/ThisInTemplateSniff.php

```diff
@@ -13,26 +14,43 @@
  */
 class ThisInTemplateSniff implements Sniff
 {
+    /**
+     * Warning violation code.
+     *
+     * @var string
+     */
+    protected $warningCodeFoundHelper = 'FoundHelper';
+
     /**
      * String representation of warning.
      *
      * @var string
      */
-    protected $warningMessage = 'Usage of $this in template files is deprecated.';
+    protected $warningMessageFoundHelper = 'Usage of helpers in templates is discouraged.';
 
     /**
      * Warning violation code.
      *
      * @var string
      */
-    protected $warningCode = 'FoundThis';
+    protected $warningCodeFoundThis = 'FoundThis';
+
+    /**
+     * String representation of warning.
+     *
+     * @var string
+     */
+    protected $warningMessageFoundThis = 'Usage of $this in template files is deprecated.';
 
     /**
      * @inheritdoc
      */
     public function register()
     {
-        return [T_VARIABLE];
+        return [
+            T_VARIABLE,
+            T_STRING,
```

Comment:
We are looking for `$this`, so we need to subscribe to `T_VARIABLE` token only. `T_STRING` is redundant here.